### PR TITLE
Add metric tracking and logging

### PR DESCRIPTION
**Pull request by Aviously**
Improves usability by tracking and displaying metrics such as the best score and elapsed time during a hyperparameter search.

### CODE REVIEW

File: kerastuner/abstractions/display.py [hunk 1]
```diff
@@ -28,15 +28,17 @@
...
+    try:
+        class_name = get_ipython().__class__.__name__
+        if "Terminal" in class_name:
+            return False
+        else:
+            return True
+
+    except NameError:
+        return False
+IS_NOTEBOOK = is_notebook()
```
Comment (omalleyt12) [inline kerastuner/abstractions/display.py:41]: Should add two blank lines before `IS_NOTEBOOK` (2 lines b/t top-level definitions is standard)

File: kerastuner/distribute/oracle_client.py [hunk 2]
```diff
@@ -45,7 +45,9 @@ def __getattr__(self, name):
             'objective',
             'max_trials',
             'allow_new_entries',
-            'tune_new_entries'}
+            'tune_new_entries',
+            'get_trial_number',
```
Comment (omalleyt12) [inline kerastuner/distribute/oracle_client.py:49]: If we end up allowing access here, we'll have to add Protobuf methods here:

https://github.com/keras-team/keras-tuner/blob/master/kerastuner/protos/service.proto

Let's discuss offline

File: kerastuner/engine/base_tuner.py [hunk 3]
```diff
@@ -287,8 +287,6 @@ def results_summary(self, num_trials=10):
         Args:
             num_trials (int, optional): Number of trials to display.
                 Defaults to 10.
-            sort_metric (str, optional): Sorting metric, when not specified
-                sort models by objective value. Defaults to None.
         """
         display.section('Results summary')
```
Comment (omalleyt12) [inline kerastuner/engine/base_tuner.py:291]: Does this still work?

File: kerastuner/engine/tuner_utils.py [hunk 4]
```diff
@@ -20,12 +20,16 @@
...
+import time
 
 import tensorflow as tf
 from tensorflow import keras
 
 from ..abstractions import display
 
+IS_NOTEBOOK = display.is_notebook()
+if IS_NOTEBOOK:
+    from IPython import display
```
Comment (omalleyt12) [inline kerastuner/engine/tuner_utils.py:32]: This seems to be overriding the same `display` from above

File: kerastuner/engine/tuner_utils.py [hunk 5]
```diff
@@ -94,17 +98,65 @@ def on_epoch_end(self, epoch, logs=None):
...
         self.verbose = verbose
+        self.oracle = oracle
 
     def on_trial_begin(self, trial):
         if self.verbose >= 1:
-            display.section('Starting new trial')
+            print()
+            trial_number = self.oracle.get_trial_number(trial)
+            total_trials = self.oracle.max_trials or '?'
+            print('Search: Running Trial {}/{}'.format(trial_number, total_trials))
```
Comment (omalleyt12) [inline kerastuner/engine/tuner_utils.py:110]: Let's just show the number of Trials this worker has complete, by keeping a running tally in tuner_utils.Display

File: kerastuner/engine/tuner_utils.py [hunk 6]
```diff
@@ -94,17 +98,65 @@ def on_epoch_end(self, epoch, logs=None):
...
+            else:
+                print() # Separate with a newline
+
+            trial_number = self.oracle.get_trial_number(trial)
+            total_trials = self.oracle.max_trials or '?'
+
+            time_taken_str = self.format_time(time.time() - self.trial_start)
+            print('Trial {}/{} Complete [{}]'.format(trial_number, total_trials, time_taken_str))
+
+            if trial.score is not None:
```
Comment (omalleyt12) [inline kerastuner/engine/tuner_utils.py:143]: Let's move this below the "Best so far" part, and maybe call it something like "Last val_accuracy:"

File: kerastuner/engine/tuner_utils.py [hunk 7]
```diff
@@ -94,17 +98,65 @@ def on_epoch_end(self, epoch, logs=None):
...
+                print('{}: {}'.format(self.oracle.objective.name, trial.score))
+
+            best_trials = self.oracle.get_best_trials()
+            if len(best_trials) > 0:
+                best_score = best_trials[0].score
+            else:
+                best_score = None
+            print('Best {} So Far: {}'.format(self.oracle.objective.name, best_score))
+
+            time_remaining = self.oracle.get_time_remaining()
```
Comment (omalleyt12) [inline kerastuner/engine/tuner_utils.py:153]: Let's just show the time the last trial took, and the total time since starting. Getting the total number of trials remaining from the chief Oracle adds too much complexity IMO

File: kerastuner/engine/base_tuner.py (hunk 3 above)
Comment (Aviously) [reply kerastuner/engine/base_tuner.py]: Deleted the docstring because it looks like `sort_metric` isn't used anymore.

File: kerastuner/abstractions/display.py [hunk 8]
```diff
@@ -1,452 +0,0 @@
...
-
-"""Display utilities."""
-
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-
-from terminaltables import SingleTable, AsciiTable
-from tabulate import tabulate
-from colorama import init, Fore, Back, Style
```
Comment (Ark-kun) [inline kerastuner/abstractions/display.py:25]: Is colorama still used anywhere?
